src/index.js

```diff
@@ -1,7 +1,180 @@
+    await $.post(
+      `${process.env.BaseURL}comments`,
+      {
+        item_id: mealDetails.idMeal,
+        username: user,
+        comment,
+      },
+      (data, status) => {
+        if (status === 'success') {
+          e.target.elements.fname.value = '';
+          e.target.elements.newUserComment.value = '';
+          getComments(mealDetails.idMeal);
+        }
+      },
+    );
+  });
+
+  popupSection.innerHTML = '';
+  popupSection.append(popupDetails);
+  popupSection.style.display = 'block';
+  getComments(mealDetails.idMeal);
+};
+
+const getComments = async (id) => {
+  await $.get(
+    `${process.env.BaseURL}comments?item_id=${id}`,
+    (data, status) => {
+      if (status === 'success') {
+        const commentDiv = document.querySelector('.comments-div');
+        commentDiv.innerHTML = '';
+        data.forEach((el) => {
+          const commentParagraph = document.createElement('p');
+          commentParagraph.innerHTML = `<span>${el.creation_date}</span><span>${el.username}</span><span>${el.comment}</span>`;
+          commentDiv.appendChild(commentParagraph);
+        });
+        countComments(data);
+      }
+    },
+  );
+};
+
+const loadPopUp = async (id) => {
+  await $.get(
+    `${process.env.URL}lookup.php?i=${id}`,
+    (data, status) => {
+      if (status === 'success') {
+        createPopHtml(data.meals[0]);
+      }
+    },
+  );
+};
+
+const innerData = (meals) => {
+  const holder = document.createElement('div');
+  holder.classList.add('grid-container');
+  meals.forEach((meal) => {
+    const div = document.createElement('div');
+    div.innerHTML = `
+      <div class="grid-item">
+          <div class="food">
+              <div>
+                 <span style="display: none" id="idCategory">${meal.idMeal}</span>
+                  <img src="${meal.strMealThumb}" alt="">
+              </div>
+              <div class="food-name padded">
+                  <span>${meal.strMeal}</span>
+                  <span class="like"><img src="${ht2}" alt="" width="15" height="15" srcset=""></span>
+              </div>
+              <div class="likes padded">
+                  <span class="likes-count">0<span> likes</span></span>
+              </div>
+              <div>
+               <input type="button" value="comments">
+              </div>
+          </div>
+      </div>`;
+    div.querySelector('input[type="button"]').addEventListener('click', () => {
+      loadPopUp(meal.idMeal);
+    });
+    div.querySelector('.like').addEventListener('click', (e) => {
+      addlikes(e.target.parentNode.parentNode.parentNode.querySelector('#idCategory').innerText);
+    });
+    holder.appendChild(div);
+  });
+  return holder;
+};
+
+const addlikes = async (itemId) => {
+  await $.post(
+    `${process.env.BaseURL}likes/`,
+    {
+      item_id: itemId,
+    },
+    (data, status) => {
+      if (status === 'success') {
+        load();
+      }
+    },
+  );
+};
+
```

Comment:
- [ ] Hi you really did a great job the only issue is that when we like a product after the request is sent and the data is updated, there is another request done to fetch the products again and it is being added to the previous ones ( when we like a product, 32 products are being added below).
 ### suggestion
 either you replace the original products with the newly fetched ones, or you stop fetching the products everytime a like is being added